### PR TITLE
DM-37276: Add skeleton configuration for live data on USDF /repo/embargo

### PR DESCRIPTION
**Pull request by gpdf**
@andy-slac Please review for any misunderstandings, etc.

@frossie I need the final alias for the /repo/embargo repository, for the DataLink URI.

### CODE REVIEW

**Comment by timj** [comment]
The build problems relate to sqlalchemy 2 and daf_relation. Fixed in lsst/daf_relation#3 but needs the requirements.txt in this package to explicitly require the github repo when building.

**Comment by timj** [comment]
You might have to pin sqlalchemy<2 in requirements.txt here because it looks like daf_butler is also broken with sqlalchemy2.

**Review by ktlim — APPROVED** [review]
I looked at the configuration documentation, but I'm not sure I have a deep understanding of what correct values would be.  Overall, this looks plausible.

**Comment by andy-slac** [comment]
There is something besides sqlalchemy 2 and due to daf_relation changes in butler that causes issues here (we do not build this package frequently, so I did not notice that). I'll make a ticket to fix that and will rebase this branch.

**Comment by andy-slac** [comment]
I rebased it to new `main` which includes DM-37771, will see if it fixes build problems.

File: configs/usdf-embargo-live.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# Configuration for the "live" ObsTAP service for /repo/embargo at USDF
+#
+# namespace and version are for possible schema migration only, do not affect table contents
+namespace: embargo
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply configs/usdf-embargo-live.yaml:4]: This namespace is not the same as Postgres schema name, more like `namespace` key in `dimensions.yaml`. It is only used for migration, I think it's OK to use `embargo`, as it reflects a known repository name.

File: configs/usdf-embargo-live.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  # We likely will need to extend this list to support most popular queries,
+  # would be good to have a list of possible queries generated by TAP.
+  instrument_name_idx: instrument_name
+  lsst_visit_idx: lsst_visit
+  lsst_exposure_idx: lsst_exposure
+  dataproduct_idx: [dataproduct_type, dataproduct_subtype]
+spectral_ranges:
+  # This list includes every band defined now in registry, actual values for some
+  # of them are probably very approximate. Keys in this section could be a band
+  # name of a physical filter name
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply configs/usdf-embargo-live.yaml]: That was typo in an example that I gave to Gregory, fixed.

**Review by andy-slac — APPROVED** [review]
Looks good to me for initial version.

File: configs/usdf-embargo-live.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+obs_collection: LATISS_LIVE       # may evolve as experience with this builds up
+collection_type: RUN              # means we are using all RUN-type collections that match line below
+collections: ["LATISS/raw/all", "LATISS/runs/quickLook/202.*"]
+use_butler_uri: false             # do not use URI from Butler, use datalink_url_fmt defined below
+dataset_types:
+  raw:
+    dataproduct_type: image
+    dataproduct_subtype: lsst.raw
+    calib_level: 1
+    obs_id_fmt: "{records[exposure].obs_id}-{records[detector].full_name}"
```
[the remark this answers is not in the document]
Comment (gpdf) [reply configs/usdf-embargo-live.yaml:17]: The `obs_id` values for DP0.2 look like this: `698740-R24_S11`, so it seems like we're following that pattern.

Do you have an alternate source for existing art that uses the "_"?

**Comment by gpdf** [comment]
Updated to use the "butler alias format" for the ID for the DataLink links service.  This should simplify deploying that service.

@ktlim provided the appropriate alias, which for this initial test is "LATISS" both at the Summit and at the USDF.

**Comment by gpdf** [comment]
I do _not_ understand why `black` is now unhappy with the Python code after my commit that changed only a YAML config.

The failed run has two extra lines in the `black` section of the action steps:

```
line_length: 110
target_version: ['py38']
```

That seems puzzling.

File: configs/usdf-embargo-live.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ktlim) [reply configs/usdf-embargo-live.yaml:17]: If this is not related to the image naming convention, then anything is fine.

**Comment by ktlim** [comment]
It's because you were lucky enough to get black 23 rather than 22.

**Comment by andy-slac** [comment]
> It's because you were lucky enough to get black 23 rather than 22.

We should probably pin 22 for now (or forever). I have one other PR broken because of this, and I expect many more to follow.

**Comment by gpdf** [comment]
OK, that did the trick.  Let's just move forward with this - that doesn't mean we shouldn't consider reverting `black`.

**Pull request merged**